src/components/RoomView/Clue.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import { setCurrentClue } from '../../store/clueSlice';
+import { useDispatch } from 'react-redux';
+import { database } from '../../utils/firebase';
+import { ref, child, push, update } from 'firebase/database';
+import { useSelector } from 'react-redux';
+import pluralize from 'pluralize';
+import { Button } from '@mui/material';
+const Clue = () => {
+  const [clueString, setClueString] = useState('');
+  const [clueNumber, setClueNumber] = useState(null);
+  const playerId = useSelector((state) => state.player.playerId);
+  const roomId = useSelector((state) => state.player.roomId);
+  const gameStatus = useSelector((state) => state.game.status);
+  const { teamOneSpymaster } = useSelector((state) => state.teamOne);
+  const { teamTwoSpymaster } = useSelector((state) => state.teamTwo);
+  let gameRef = ref(database, 'rooms/' + roomId + '/game/');
+  //get all words in game
+  const gameboard = useSelector((state) => state.wordsInGame.wordsInGame);
+  let arrayToCheck = [];
+  //push all words in gameboard into an array
+  for (let i = 0; i < gameboard.length; i++) {
+    arrayToCheck.push(gameboard[i].word.toUpperCase());
+  }
+  let cluesRef = ref(database, 'rooms/' + roomId + '/clues/');
+
+  const dispatch = useDispatch();
+
+  const handleClueChange = (event) => {
+    //trim any extra space so users cannot submit "   clue" or '    ',
+    //then convert all to uppercase to avoid case sensitive issue
+    setClueString(event.target.value.trim().toUpperCase());
+  };
+
+  //please also help me rephrase all these alert messages
+  const handleNumberChange = (event) => {
+    //this is to prevent users from submitting 'select a number'
+    if (event.target.value.length > 1) {
+      alert('please select a valid number');
+    } else setClueNumber(Number(event.target.value));
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const regex = /[\s-]+/g;
+    if (clueString === '') {
+      return alert('you have to enter a clue');
+    }
+    //then convert all to singular so users cannot use 'apples' or 'feet' to indicate 'apple','foot'
+    //not sure how accurate .singular() is tho
+    else if (arrayToCheck.includes(clueString) || arrayToCheck.includes(pluralize.singular(clueString))) {
+      return alert('this word is already on the gameboard, you cannot use it as your clue');
+    }
+    //here we make rules to only allow compound word that is a union of 3(and less) words so people won't type a whole sentence,
+    // i.e "New York" or 'mother-in-law'
+    else if (clueString.match(regex) && clueString.match(regex).length > 2) {
+      return alert('you can only use a compound word that is made of less than 3 words');
+    } else {
+      const clueData = {
+        clueString,
+        clueNumber,
```

Comment:
i noticed after playing around with this that we're currently allowing the player to not select a number. do you know from the real game rules what the default is? or should we just require thjat they select a number?

Reply:
oh that's a thing I missed, will add the default value in a bit